checks.d/postgres.py

```diff
@@ -334,22 +354,31 @@ def _collect_stats(self, key, db, instance_tags, relations, custom_metrics):
         If custom_metrics is not an empty list, gather custom metrics defined in postgres.yaml
         """
 
-        self.DB_METRICS['metrics'] = self._get_instance_metrics(key, db)
-        self.BGW_METRICS['metrics'] = self._get_bgw_metrics(key, db)
         metric_scope = [
-            self.DB_METRICS,
             self.CONNECTION_METRICS,
-            self.BGW_METRICS,
             self.LOCK_METRICS,
-            self.COUNT_METRICS
+            self.COUNT_METRICS,
         ]
 
+        # These are added only once per PG server, thus the test
+        db_instance_metrics = self._get_instance_metrics(key, db)
+        bgw_instance_metrics = self._get_bgw_metrics(key, db)
+
+        if db_instance_metrics is not None:
+            self.DB_METRICS['metrics'] = db_instance_metrics
```

Comment:
It makes me sad to see what should be a constant being muted :( we should clean that in a next version of this check